### PR TITLE
Removes the unneeded neural network for AI movement

### PR DESCRIPTION
**Pull request by Bod9001**
turns out the neural network was only Used for AI movement, that's pretty much all it was used for, 
this has been tested with xeno's and face huggers

### CODE REVIEW

File: UnityProject/Assets/Scripts/NPC/AI/MobObjective.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+
+	public virtual void ContemplatePriority()
+	{
+	}
+
+
+	public void Move(Vector3Int dirToMove)
+	{
+		var dest = MobTile.LocalPositionServer + (Vector3Int)dirToMove;
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/AI/MobObjective.cs]:
```suggestion
		var dest = MobTile.LocalPositionServer + dirToMove;
```

File: UnityProject/Assets/Scripts/NPC/AI/MobMeleeAction.cs [hunk 2]
```diff
@@ -266,41 +254,45 @@ protected virtual bool PerformActionNpc(MatrixManager.CustomPhysicsHit hitInfo,
...
 			if (TargetDistance() > 4.5f)
 			{
 				//Don't bother, the target is too far away to warrant a decision to break a tile
-				return false;
+				return;
 			}
 
 			ActOnTile(hitInfo.TileHitWorld.RoundToInt(), dir);
-			return true;
+			return;
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/AI/MobMeleeAction.cs]:
```suggestion
```

File: UnityProject/Assets/Scripts/NPC/AI/MobFollow.cs [hunk 3]
```diff
@@ -1,132 +1,95 @@
...
+				var Distance = TargetDistance();
+				if (Distance > 15)
+				{
+					FollowTarget = null;
+					Priority = 0;
+					return;
+				}
+				else
+				{
+
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/AI/MobFollow.cs]:
```suggestion
```

File: UnityProject/Assets/Scripts/NPC/AI/MobFollow.cs [hunk 4]
```diff
@@ -1,132 +1,95 @@
...
-			if (FollowTarget == null) return;
 
-			PerformMoveAction(Mathf.FloorToInt(vectorAction[0]));
-		}
 
-		protected override void OnPushSolid(Vector3Int destination)
+		public override void DoAction()
 		{
-			if (destination == Vector3Int.RoundToInt(TargetTile.WorldPositionServer))
+			if (FollowTarget== null) return;
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/AI/MobFollow.cs]:
```suggestion
			if (FollowTarget == null) return;
```

File: UnityProject/Assets/Scripts/NPC/AI/MobFollow.cs [hunk 5]
```diff
@@ -1,132 +1,95 @@
...
+			var moveToRelative = (FollowTarget.WorldPositionServer - MobTile.WorldPositionServer ).ToNonInt3();
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/AI/MobFollow.cs]:
```suggestion
			var moveToRelative = (FollowTarget.WorldPositionServer - MobTile.WorldPositionServer).ToNonInt3();
```

File: UnityProject/Assets/Scripts/NPC/AI/MobController.cs [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
+	private void OnEnable()
+	{
+		UpdateManager.Add(UpdateMe, 0.85f);
+	}
+
+	private void OnDisable()
+	{
+		UpdateManager.Remove( CallbackType.PERIODIC_UPDATE, UpdateMe );
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/AI/MobController.cs]:
```suggestion
		UpdateManager.Remove(CallbackType.PERIODIC_UPDATE, UpdateMe);
```

File: UnityProject/Assets/Scripts/NPC/AI/MobAgent.cs [hunk 7]
```diff
@@ -1,369 +1,14 @@
...
 namespace Systems.MobAIs
 {
-	/// <summary>
-	/// Handles the underlying logic for
-	/// the Mob[Brain] behaviours
-	/// </summary>
-	[RequireComponent(typeof(CustomNetTransform))]
-	[RequireComponent(typeof(RegisterObject))]
-	public class MobAgent : Agent, IServerLifecycle
+	public class _MobController : MonoBehaviour
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/AI/MobAgent.cs]: Empty component?

File: UnityProject/Assets/Scripts/NPC/AI/MobAI.cs [hunk 8]
```diff
@@ -209,7 +200,7 @@ private void MonitorFollowingTime()
 
 		private void MonitorExploreTime()
 		{
-			if (mobExplore.activated && exploreTimeMax > 0)
+			if (mobExplore.Priority  < 25 && exploreTimeMax > 0)
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/AI/MobAI.cs]:
```suggestion
			if (mobExplore.Priority < 25 && exploreTimeMax > 0)
```

File: UnityProject/Assets/Scripts/NPC/AI/Hostile/StatueAI.cs [hunk 9]
```diff
@@ -84,7 +84,7 @@ private bool IsSomeoneLookingAtMe()
 		protected override void MonitorIdleness()
 		{
 
-			if (!mobMeleeAction.performingDecision && mobMeleeAction.FollowTarget == null && !IsSomeoneLookingAtMe())
+			if ( mobMeleeAction.FollowTarget == null && !IsSomeoneLookingAtMe())
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/AI/Hostile/StatueAI.cs]:
```suggestion
			if (mobMeleeAction.FollowTarget == null && !IsSomeoneLookingAtMe())
```

File: UnityProject/Assets/Scripts/NPC/AI/Friendly/GenericFriendlyAI.cs [hunk 10]
```diff
@@ -43,7 +43,7 @@ protected override void UpdateMe()
 
 		protected virtual void MonitorExtras()
 		{
-			if (IsPerformingTask && !doRandomActionWhenInTask)
+			if ( !doRandomActionWhenInTask)
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/AI/Friendly/GenericFriendlyAI.cs]:
```suggestion
			if (!doRandomActionWhenInTask)
```

File: UnityProject/Assets/Scripts/NPC/AI/MobController.cs [hunk 11]
```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+public class MobController : MonoBehaviour
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/AI/MobController.cs]: Namespace to `Systems.MobAIs`, perhaps?

File: UnityProject/Assets/Scripts/NPC/AI/MobObjective.cs [hunk 12]
```diff
@@ -0,0 +1,95 @@
+using System.Collections;
+using System.Collections.Generic;
+using Doors;
+using UnityEngine;
+
+public class MobObjective : MonoBehaviour
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/AI/MobObjective.cs]: `Systems.MobAIs` here too?

**Review by ThatDan123 — APPROVED** [review]

**Review by NoooneyDude — APPROVED** [review]

**Pull request merged**